pkg/controller/egress/controller.go

```diff
@@ -274,19 +285,25 @@ func (c *EgressController) syncEgressIP(egress *egressv1beta1.Egress) (net.IP, e
 	if egress.Spec.EgressIP != "" {
 		ip = net.ParseIP(egress.Spec.EgressIP)
 		if err := c.externalIPAllocator.UpdateIPAllocation(egress.Spec.ExternalIPPool, ip); err != nil {
+			statusReason = "ExternalIPPoolError"
+			statusMessage = err.Error()
 			return nil, fmt.Errorf("error when allocating IP %v for Egress %s from ExternalIPPool %s: %v", ip, egress.Name, egress.Spec.ExternalIPPool, err)
 		}
 	} else {
 		var err error
 		// User doesn't specify the Egress IP, allocate one.
 		if ip, err = c.externalIPAllocator.AllocateIPFromPool(egress.Spec.ExternalIPPool); err != nil {
+			statusReason = "ExternalIPPoolError"
+			statusMessage = err.Error()
 			return nil, err
 		}
 		if err = c.updateEgressIP(egress, ip.String()); err != nil {
```

Comment:
`updateEgressIP` could update the `egress` in kube-apiserver and it will be returned by `Patch` function, we'd better to return the updated `egress`, otherwise using the old `egress` object to update condition will always fail once as the resouce version doesn't match.